### PR TITLE
Added PCI XROM enumeration function to PCI HAL module. Also added a P…

### PR DESCRIPTION
**Pull request by domcc3**
…CI_Util option to enumerate XROM's.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,5 @@
...
 python:
-  - "2.6"
   - "2.7"
-  - "3.2"
-  - "3.3"
-  - "3.4"
-  - "3.5"
 # command to run tests
-script: python source/tool/test_util.py
+script: python -m unittest discover tests/software/
```
Comment (tweksteen) [inline .travis.yml]: This is the correct command, you need to modify the PYTHON_PATH to include source/tools

File: source/tool/chipsec/hal/pci.py [hunk 2]
```diff
@@ -130,6 +130,22 @@ def enumerate_devices( self ):
                         devices.append( (b, d, f, vid, did) )
         return devices
 
+    def enumerate_xroms( self ):
+       	pci_devices = self.enumerate_devices()
```
Comment (tweksteen) [inline source/tool/chipsec/hal/pci.py]: Please fix the indentation so it matches the rest of the file

File: source/tool/chipsec/hal/pci.py [hunk 3]
```diff
@@ -130,6 +130,22 @@ def enumerate_devices( self ):
...
+                #read at an offset of 0x30 (the location of the XROM BAR
+                xrom_bar = self.read_dword(device[0], device[1], device[2], 0x30)
+                if logger().HAL: logger().log("BAR at B,D,F of %d, %d, %d:   0x%08x" % (device[0], device[1], device[2], xrom_bar))
+		#use the logging below if you want to read the contents of the place where the XROM's are mapped
+                #mem_addr = xrom_bar & 0xFFFFFFF0
+                #logger().log("Memory contents at address 0x%08x: 0x%08x" % (mem_addr, self.cs.mem.read_physical_mem_dword(mem_addr)))
+                if((xrom_bar & 0x00000001) == 0x01): #if the LSB is 1
+                        if logger().HAL: logger().log("PCI XROM found")
+                        xrom_devices.append(device.vid, device.did, xrom_bar)
+	logger().log("\n%d total XROM's found. You can modify hal/pci.py to print the contents of the XROM(s) if any were found.\n" % len(xrom_devices))
```
Comment (tweksteen) [inline source/tool/chipsec/hal/pci.py]: The newline characters are not necessary, please remove them

File: source/tool/chipsec/utilcmd/pci_cmd.py [hunk 4]
```diff
@@ -67,6 +68,13 @@ def pci(argv):
         logger().log( "[CHIPSEC] (pci) time elapsed %.3f" % (time.time()-t) )
         return
 
+    if ( 'xrom_enumerate' == op ):
+        logger().log( "[CHIPSEC] Enumerating PCIe device XROM's..." )
+	pci = Pci(chipsec_util._cs)
```
Comment (tweksteen) [inline source/tool/chipsec/utilcmd/pci_cmd.py]: indentation

**Comment by tweksteen** [comment]
You need to separate the commit for the options ROM and the commits for deploying Travis. Create another pull request for the Travis commits.

**Comment by c7zero** [comment]
Added OpROM enumeration but had to extend it to cover Type1 PCI headers, also initializing XROM BARs when they are not initialized with supplied MMIO address and dumping OpROM contents + parsing architectural OpROM headers (legacy vs EFI is not yet parsed)

**Pull request closed without merging**